rules/utils/assemblyinput.rule

```diff
@@ -8,15 +8,26 @@ assembly_dir = ep_params['outdir']
 basename = config['basename']
 assembly_extension = aInput_params['assembly_extension']
 
+
 rule cp_input_assembly:
     input: aInput_params['assembly']
     output: join(assembly_dir, basename + assembly_extension + '.fasta')
     log: join(LOGS_DIR, 'assemblyinput/cp_assembly.log')
-    shell: ("cp {input} {output}")
+    params: assembly_dir = ep_params['outdir']
+    shell: 
+        """
+        mkdir -p {params.assembly_dir}
```

Comment:
Given that the output dir is `ep_params['outdir']` and therefore something common to all the rules (not just assembly), I'm wondering if this mkdir command should really go here.

It might be better to ensure the output directory exists independent of which rule or workflow is being run, possibly by doing it in the `run_eelpond` script (specifically, around [line 88](https://github.com/dib-lab/eelpond/blob/bdaf28162bc6f3d5763efa4ddec03f6da89f213e/run_eelpond#L88)).

Reply:
Just delayed response because I'm not sure. Right now, `run_eelpond` doesn't make any dirs or do anything of substance - it just aggregates and builds _paths_ for directories (snakemake takes care of actually building everything while running the rules themselves).

Reply:
Thinking about it now, `get_data` should also fail when using `out_path`, if the path does not already exist.

Reply:
Ready for review now @charlesreid1! This PR solves the `out_path` issue, but does not yet implement separated high-level input/output checking, as discussed in #110. I left a note in for the things I think should move into a fix for #110, but I'd like to implement that in a separate PR so that these fixes can get integrated.